### ISSUE TITLE
Whenever a search query is entered, the whole table is cleared

### ISSUE THREAD

**Issue by AlexandreBonneau** [xxRockOnxx/vue-flexi-table#2]
## What happens
Using https://xxrockonxx.github.io/vue-flexi-table/demo/index.html#/, whenever you enter any search text (ie. `del`), all the table rows are deleted, then the filtered results are shown.

## Expected result
The table rows should not be modified while the search text is modified. It should change only when new table rows are available.

**Comment by xxRockOnxx** [comment]
Uhm, not sure if I understand your question correctly. Shouldn't it be working that way? Same way as to how [datatables](https://datatables.net/) work? Try searching there also (ie. `airi`)

**Comment by AlexandreBonneau** [comment]
Well, that `datatables` plugin works great ; in that example, if you enter just `'a'`, then `'ai'`, the rows are *always* shown, while with `vue-flexi-table`, when you type `'a'`, 2 thing happen very quickly:
1. All the rows are deleted
2. Then the filtered rows are populated back

The rows should not be deleted first, only filtered.

**Comment by xxRockOnxx** [comment]
Ah, it's because of the `debounce` I included. Do you suggest that I remove it instead? It is set for `250ms` currently. Personally I like it that way.

**Comment by AlexandreBonneau** [comment]
The fact that the query is debounced is not relevant here I think ; it's just that while you are typing the search term, the table is emptied.

No matter how long you wait for filtering the data, it should not first remove all the table content.

**Comment by xxRockOnxx** [comment]
It really is the `debounce`. The loading status will turn to `false` after the searching is done which currently is being `debounced`.

In short:
- Once you typed: loading is `true`
- After searching: loading will become `false`

By the nature of `debounce`, searching will only finish if you stopped typing after `x` seconds.

I think the issue is subjective but you have a good reason. Maybe a loading div in the middle of the table will do? Or do you suggest anything? I don't like it if there's no any loading status indicator, it just makes it look like unresponsive.

**Comment by AlexandreBonneau** [comment]
Take a look at this [tweet](https://twitter.com/youyuxi/status/857485825868578818) from Evan, there is an idea there with the delay before showing a loading image.

I still think you should *not* clear the table until the new data comes.

For instance, what happens if you start to type something, you clear the table, but then somehow the response never come? Do you end up with an empty table?

**Comment by xxRockOnxx** [comment]
Yes, again, personally I liked it to respond right away when I typed something so I was a bit bias there and also, I guess I learned something new today.

Ok, it is agreed. On the next release I will try to move or implement a different indicator. If you have a suggestion that would be great also. This is a side-project so, important ones comes first.
